plugin-test/Categorifier/Test/HList.hs

```diff
@@ -59,3 +66,38 @@ zipWith fn (HCons1 hf tf) (HCons1 hg tg) = HCons1 (fn hf hg) $ zipWith fn tf tg
 zipLowerWith :: (forall a. f a -> g a -> c) -> HList1 f l -> HList1 g l -> Vec (Length l) c
 zipLowerWith _ HNil1 HNil1 = VNil
 zipLowerWith fn (HCons1 hf tf) (HCons1 hg tg) = fn hf hg ::: zipLowerWith fn tf tg
+
+data HMap1 (f :: v -> Type) (l :: [(k, v)]) where
+  HEmpty1 :: HMap1 f '[]
+  HInsert1 :: Proxy k -> f e -> HMap1 f l -> HMap1 f ('(k, e) ': l)
+
+type family Value l k :: Type where
+  Value '[] k = Void
+  Value ('(k, v) ': t) k = v
+  Value ('(k', v) ': t) k = Value t k
+
+data HeadMatches
+  = HeadMatches
+  | HeadDoesntMatch HeadMatches
+
+class HMapOps f l k headmatches where
```

Comment:
Where is this class used, and what's `headMatches` for?

Reply:
This is also part of the stuff that shouldn't have been committed.

It should all be gone now.